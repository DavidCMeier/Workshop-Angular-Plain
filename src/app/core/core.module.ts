import { StoreModule } from "@ngrx/store";
import { EffectsModule } from "@ngrx/effects";
import * as fromStore from './store';
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { HeaderComponent } from './components/header/header.component';
import { FooterComponent } from './components/footer/footer.component';
import { ShellComponent } from './containers/shell/shell.component';
import { MatToolbarModule } from "@angular/material/toolbar";
import { RouterModule } from "@angular/router";
import { MatSelectModule } from "@angular/material/select";
import { MatIconModule } from "@angular/material/icon";
import { MatButtonModule } from "@angular/material/button";
import { HTTP_INTERCEPTORS, HttpClientModule } from "@angular/common/http";
import { AuthInterceptor } from "./interceptors/auth-interceptor.service";
import { SharedModule } from "../shared/shared.module";
import { StoreRouterConnectingModule } from "@ngrx/router-store";



@NgModule({
  declarations: [
    HeaderComponent,
    FooterComponent,
    ShellComponent
  ],
  imports: [
    StoreModule.forFeature('core', fromStore.reducers),
    EffectsModule.forFeature(fromStore.effects),
    StoreRouterConnectingModule.forRoot(),
    CommonModule,
    MatToolbarModule,
    RouterModule,
    MatSelectModule,
    MatIconModule,
    MatButtonModule,
    HttpClientModule,
    SharedModule
  ],
  providers: [
    { provide: HTTP_INTERCEPTORS, useClass: AuthInterceptor, multi: true }
  ],
})
export class CoreModule { }
