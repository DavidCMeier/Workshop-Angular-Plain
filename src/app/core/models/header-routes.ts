export interface HeaderRoutes {
  name: string;
  path: string;
}
