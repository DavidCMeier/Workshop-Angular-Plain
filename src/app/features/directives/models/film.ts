export interface Film {
  id: string;
  title: string;
  director: string;
  year: string;
  image: string;
  description?: string;
}

export interface FilmApiResponse {
  id: string,
  title: string,
  description: string,
  director: string,
  release_date: string,
  image: string,
}
