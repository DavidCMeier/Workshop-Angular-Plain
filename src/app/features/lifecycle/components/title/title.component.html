  <h1 #titleElement [appSizeText]="size">{{title}}</h1>
