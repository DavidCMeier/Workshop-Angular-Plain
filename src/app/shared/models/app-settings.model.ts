export interface AppSettings{
  production: boolean;
  apiUrl: string;
}
