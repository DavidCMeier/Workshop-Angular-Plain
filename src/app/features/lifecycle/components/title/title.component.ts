import {
  AfterViewInit,
  Component,
  ElementRef,
  Input,
  OnChanges,
  OnDestroy,
  OnInit,
  SimpleChanges,
  ViewChild,
  ViewEncapsulation
} from '@angular/core';
import { Size } from "../../../../shared/models/size";

@Component({
  selector: 'app-title',
  templateUrl: './title.component.html',
  styleUrls: ['./title.component.scss'],
  encapsulation: ViewEncapsulation.None
})
export class TitleComponent implements OnInit, OnChanges, AfterViewInit, OnDestroy {

  @Input() title!: string;
  @Input() size!: Size;
  @ViewChild('titleElement') titleElement!: ElementRef;
  constructor() {
    console.log('Constructor', this.title, this.size, this.titleElement);
  }

  ngOnInit(): void {
    console.log('OnInit', this.title, this.size, this.titleElement);
  }

  ngAfterViewInit(): void {
    console.log('AfterViewInit', this.title, this.size, this.titleElement);
  }

  ngOnChanges(changes: SimpleChanges): void {
    console.log('OnChanges',changes);
  }

  ngOnDestroy(): void {
    console.log('OnDestroy');
  }

}
