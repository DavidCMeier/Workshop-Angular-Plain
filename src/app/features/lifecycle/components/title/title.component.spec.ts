import { ComponentFixture, TestBed } from '@angular/core/testing';

import { TitleComponent } from './title.component';
import { SizeTextDirective } from "../../../../shared/directives/size-text.directive";
import { Size } from "../../../../shared/models/size";

describe('TitleComponent', () => {
  let component: TitleComponent;
  let fixture: ComponentFixture<TitleComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [ TitleComponent, SizeTextDirective ]
    })
    .compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(TitleComponent);
    component = fixture.componentInstance;
    component.title = 'Test title';
    component.size = Size.LARGE;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
    expect(fixture).toMatchSnapshot();
  });

});
