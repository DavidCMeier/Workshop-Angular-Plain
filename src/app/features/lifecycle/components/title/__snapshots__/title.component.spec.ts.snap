// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleComponent should be render with a diferent Size 1`] = `
<app-title
  __ngContext__={[Function LRootView]}
  size={[Function String]}
  title={[Function String]}
  titleElement={[Function ElementRef]}
>
  <h1>
    Test title
  </h1>
</app-title>
`;

exports[`TitleComponent should create 1`] = `
<app-title
  __ngContext__={[Function LRootView]}
  size={[Function String]}
  title={[Function String]}
  titleElement={[Function ElementRef]}
>
  <h1
    style="font-size: 1.4rem;"
  >
    Test title
  </h1>
</app-title>
`;
